src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -16,16 +16,30 @@ Status ServiceBasedJobInfoAccessor::AsyncAdd(
                  << ", driver pid = " << data_ptr->driver_pid();
   rpc::AddJobRequest request;
   request.mutable_data()->CopyFrom(*data_ptr);
-  client_impl_->GetGcsRpcClient().AddJob(
-      request,
-      [job_id, data_ptr, callback](const Status &status, const rpc::AddJobReply &reply) {
-        if (callback) {
-          callback(status);
-        }
-        RAY_LOG(DEBUG) << "Finished adding job, status = " << status
-                       << ", job id = " << job_id
-                       << ", driver pid = " << data_ptr->driver_pid();
-      });
+
+  Executor* executor = new Executor(client_impl_);
```

Comment:
use smart pointers?

Reply:
Executor will be delete if rpc operation finished. If rpc operation failed,  we will use executor retry.